### PR TITLE
added code validation and packaging via gulp

### PR DESCRIPTION
**Pull request by pulsar256**
check the README.md file in the "support/" directory for documentation.

### CODE REVIEW

File: support/README.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+the default task will perform all validation steps and package the application. For Windows users it is recommended to use the Cygwin-Shell as the `.jshintrc` file will not be picked up correctly otherwise resulting in too strict validation and lots of false alarms.
+
+### Prerequisites
+
+* node.js (+npm)
+* Cygwin bash/zsh shell (Windows users only)
+
+### Installing / Running
+
+Install all npm dev dependencies and optionally update your PATH (sentenv.sh) environment variable if you do not plan on installing gulp globally.
```
Comment (hydra) [inline support/README.md]: `setenv.sh`

**Comment by hydra** [comment]
cool i'll give this a whirl asap

**Comment by hydra** [comment]
not had the chance to look yet.  gonna merge some more stuff into v1.12.0 firmware and then work on the configurator more.  likely will merge this before next configurator release.

**Comment by NightHawk32** [comment]
Gave this a try,
very good idea @pulsar256.
we should merge this to an own branch since there are a lot of issues and warnings reported. We could handle this by just adding the travis.yml only to that special branch and by setting "Build only if .travis.yml is present" in travis ci.
This only becomes usable when all the issues are resolved, otherwise you won't be able to tell if a commit produced any new issues.
Solving all problems will take some time.

**Comment by hydra** [comment]
we can still use it but not cause builds to fail right now.

@NightHawk32 @skaman82 If you could start work on fixing all the issues reported in the mean time that would be great.

**Comment by skaman82** [comment]
@hydra I took a look the css warnings and took care of the most of them in the last PR but couldn't remove all of them. Warnings like rgba, float and !importaint issues should be ignored.


Von meinem iPhone gesendet

**Comment by hydra** [comment]
I'm going to close this PR so we can track current issues.  This PR and conversation can serve as a reference for future discussions.

@pulsar256 @skaman82 @NightHawk32 Thanks very much for your work on this, we can look at this again once we figure out a direction for the configurator.

**Pull request closed without merging**